File: Application/Dto/Users/TeacherDto.cs
using Application.Dto.School;
using Domain.Entities_v2.Types;

namespace Application.Dto.Users;

public class TeacherDto: IBaseDto
{
    public int Id { get; set; }
    public DateTime Created { get; set; }
    public DateTime Updated { get; set; }
    public Status Status { get; set; }
    public string Login { get; set; } = "";
    public string FirstName { get; set; } = "";
    public string LastName { get; set; } = "";
    public string Phone { get; set; } = "";
    public string Role { get; set; } = "";
    public DateTime BirthDate { get; set; }
    public AddressDto Address { get; set; } 
    public int SchoolId { get; set; }
}

File: frontend/src/Domain/Dtos/TeacherDto.ts
import AddressDto from '@/domain/dtos/AddressDto.ts';
import EntityBase from '@/domain/EntityBase.ts';

interface TeacherDto extends EntityBase {
  login: string;
  firstName: string;
  lastName: string;
  phone: string;
  role: string;
  birthDate: string;
  address: AddressDto;
  schoolId: number;
}

export default TeacherDto;
